src/CameraAravis.cpp

```diff
@@ -159,7 +165,7 @@ namespace camera
             new_frame.swap(frame);
             new_frame.init(frame.size.width,frame.size.height,8,frame.frame_mode,-1); // ensure right size
             frame.setStatus(base::samples::frame::STATUS_VALID);
-            frame.time = base::Time::fromMicroseconds(arv_buffer_get_timestamp(arv_buffer)*0.001);
+            frame.time = start_time + base::Time::fromMicroseconds(arv_buffer_get_timestamp(arv_buffer)*0.001);
```

Comment:
Multiplying by 0.001 does a conversion to floating-point, which is not necessary and can lead to loss of precision. Since we're dealing with integers in and out, just divide by 1000.